mage_ai/frontend/pages/triggers/index.tsx

```diff
@@ -37,10 +42,35 @@ function TriggerListPage() {
     [project?.features],
   );
 
+  // Debounce search input
+  useEffect(() => {
+    if (debounceTimeout.current) {
+      clearTimeout(debounceTimeout.current);
+    }
+
+    debounceTimeout.current = setTimeout(() => {
+      setSearchQuery(searchText);
+    }, 500);
+
+    return () => {
+      if (debounceTimeout.current) {
+        clearTimeout(debounceTimeout.current);
+      }
+    };
+  }, [searchText]);
+
+  // Reset page to 0 when search query changes
+  useEffect(() => {
```

Comment:
If you remove the auto-fetch when user types in the keyboard, this useEffect shouldn't be necessary either.